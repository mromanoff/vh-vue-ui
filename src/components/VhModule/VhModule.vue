<template>
  <!-- VhMN-3948 temporary nuxt-link override -->
  <div
    :id="this.anchorId"
    class="VhModule"
    :class="{ [`VhModule--${this.size}`]: this.size, 'VhModule--anchor': this.anchorId }"
  >
    <slot />
  </div>
</template>

<script>
export default {
  name: 'VhModule',
  /**
   *  One of these size: small, medium, large
   */
  props: {
    anchorId: {
      type: String,
      default: null,
    },
    size: {
      type: String,
      default: 'medium',
    },
  },
};
</script>

<style scoped>
.VhModule {
  /* BLOCK */
}

.VhModule:first-child {
  margin-top: 0;
}

.VhModule--small {
  margin-top: var(--VhModule--mobile-margin-small);
  margin-bottom: var(--VhModule--mobile-margin-small);
}

.VhModule--medium {
  margin-top: var(--VhModule--mobile-margin-medium);
  margin-bottom: var(--VhModule--mobile-margin-medium);
}

.VhModule--large {
  margin-top: var(--VhModule--mobile-margin-large);
  margin-bottom: var(--VhModule--mobile-margin-large);
}

/* VhMN-3948 temporary nuxt-link override */
.VhModule--small.VhModule--anchor {
  padding-top: calc(var(--space--medium) + var(--Header--mobile-height));
  margin-top: calc(var(--VhModule--mobile-margin-small) * -1);
}

.VhModule--medium.VhModule--anchor {
  padding-top: calc(var(--space--medium) + var(--Header--mobile-height));
  margin-top: calc(var(--VhModule--mobile-margin-medium) * -1);
}

.VhModule--large.VhModule--anchor {
  padding-top: calc(var(--space--medium) + var(--Header--mobile-height));
  margin-top: calc(var(--VhModule--mobile-margin-large) * -1);
}

@media (--viewport--medium-up) {
  .VhModule--small {
    margin-top: var(--VhModule--desktop-margin-small);
    margin-bottom: var(--VhModule--desktop-margin-small);
  }

  .VhModule--medium {
    margin-top: var(--VhModule--desktop-margin-medium);
    margin-bottom: var(--VhModule--desktop-margin-medium);
  }

  .VhModule--large {
    margin-top: var(--VhModule--desktop-margin-large);
    margin-bottom: var(--VhModule--desktop-margin-large);
  }
}
</style>
