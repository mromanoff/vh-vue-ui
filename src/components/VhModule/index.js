import VhModule from './VhModule';

export default VhModule;
