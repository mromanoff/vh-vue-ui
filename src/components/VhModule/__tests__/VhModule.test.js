import { shallowMount } from '@vue/test-utils';
import VhModule from '../VhModule.vue';

describe('VhModule.vue', () => {
  let wrapper;

  it('renders default VhModule', () => {
    wrapper = shallowMount(VhModule);
    expect(wrapper.classes()).toContain('VhModule--medium');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders large VhModule', () => {
    wrapper = shallowMount(VhModule, {
      propsData: {
        size: 'large',
      },
    });
    expect(wrapper.classes()).toContain('VhModule--large');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders small VhModule', () => {
    wrapper = shallowMount(VhModule, {
      propsData: {
        size: 'small',
      },
    });
    expect(wrapper.classes()).toContain('VhModule--small');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders VhModule with anchor', () => {
    wrapper = shallowMount(VhModule, {
      propsData: {
        anchorId: 'testId',
      },
    });
    expect(wrapper.classes()).toContain('VhModule--anchor');
    expect(wrapper.attributes().id).toBe('testId');
    expect(wrapper).toMatchSnapshot();
  });
});
