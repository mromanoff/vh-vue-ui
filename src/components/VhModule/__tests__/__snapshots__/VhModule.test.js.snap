// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VhModule.vue renders VhModule with anchor 1`] = `<div id="testId" class="VhModule VhModule--medium VhModule--anchor"></div>`;

exports[`VhModule.vue renders default VhModule 1`] = `<div class="VhModule VhModule--medium"></div>`;

exports[`VhModule.vue renders large VhModule 1`] = `<div class="VhModule VhModule--large"></div>`;

exports[`VhModule.vue renders small VhModule 1`] = `<div class="VhModule VhModule--small"></div>`;
